clients/privacy-center/packages/fides-consent/src/lib/consent-banner.ts

```diff
@@ -0,0 +1,553 @@
+}
+
+div#fides-consent-banner.fides-consent-banner-bottom.fides-consent-banner-hidden {
+  transform: translateY(100%);
+}
+
+div#fides-consent-banner.fides-consent-banner-top {
+  position: fixed;
+  z-index: 1;
+  width: 100%;
+  top: 0;
+  left: 0;
+  transform: translateY(0%);
+}
+
+div#fides-consent-banner.fides-consent-banner-top.fides-consent-banner-hidden {
+  transform: translateY(-100%);
+}
+
+
+div#fides-consent-banner-description {
+  margin-top: 0.5em;
+  margin-right: 2em;
+  min-width: 33%;
+  flex: 1;
+}
+
+div#fides-consent-banner-buttons {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+}
+
+button.fides-consent-banner-button {
+  display: inline-block;
+  flex: auto;
+  cursor: pointer;
+  text-align: center;
+  margin: 0;
+  margin-top: 0.25em;
+  margin-right: 0.25em;
+  padding: var(--fides-consent-banner-button-padding);
+  background: var(--fides-consent-banner-button-background);
+  color: var(--fides-consent-banner-button-text-color);
+  border: none;
+  border-radius: var(--fides-consent-banner-button-border-radius);
+
+  font-family: inherit;
+  font-size: 100%;
+  line-height: 1.15;
+  text-decoration: none;
+}
+
+button.fides-consent-banner-button:hover {
+  background: var(--fides-consent-banner-button-hover-background);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-primary {
+  background: var(--fides-consent-banner-button-primary-background);
+  color: var(--fides-consent-banner-button-primary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-primary:hover {
+  background: var(--fides-consent-banner-button-primary-hover-background);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-secondary {
+  background: var(--fides-consent-banner-button-secondary-background);
+  color: var(--fides-consent-banner-button-secondary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-secondary:hover {
+  background: var(--fides-consent-banner-button-secondary-hover-background);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-tertiary {
+  background: var(--fides-consent-banner-button-tertiary-background);
+  color: var(--fides-consent-banner-button-tertiary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-tertiary:hover {
+  background: var(--fides-consent-banner-button-tertiary-hover-background);
+}
+`
+
+/**
+ * Builds a 'style' element containing the CSS styles for the consent banner
+ */
+const buildStyles = (): Node => {
+  const style = document.createElement("style");
+  style.innerHTML = globalBannerStyle;
+  return style;
+};
+
+/**
+ * Initialize the Fides Consent Banner, with optional extraOptions to override defaults.
+ * 
+ * (see the type definition of ConsentBannerOptions for what options are available)
+ */
+export const initBanner = async (defaults: CookieKeyConsent, extraOptions?: ConsentBannerOptions): Promise<void> => {
```

Comment:
Reviewer note: bringing it all together!

Note that this is an `async` function as initialization can't be assumed to be synchronous - in particular, it'll require a geolocation API to finalize. This has implications on how you integrate the banner into a website too - since `Fides.banner()` isn't guaranteed to show the banner yet, we'll need to add some kind of callback functions like `onBannerInitialized()` to let users know when it's safe to start querying the `Fides.consent` values and triggering other code that should be dependent on consent preferences.